File: Miscellaneous/leap_year.cpp
// Write Java Program to find out whether the year is a leap year
#include<bits/stdc++.h>
using namespace std;

class Solution {
    public : 
    void leapYear(int year) {
        if (year % 4 == 0) {
        if (year % 100 == 0) {
            if (year % 400 == 0)
                cout << year << " is a leap year.";
            else
                cout << year << " is not a leap year.";
        }
        else
            cout << year << " is a leap year.";
    }
    else
        cout << year << " is not a leap year.";
    }
};

int main(){
    int year;
    cin >> year;

    Solution obj;
    obj.leapYear(year); 

    return 0;
}  

File: Miscellaneous/leap_year.java
import java.util.Scanner;

public class leap_year {
    public static void leapYear(int year) {
        if (year % 4 == 0) {
        if (year % 100 == 0) {
            if (year % 400 == 0)
                System.out.println(year + " is a leap year.");
            else
                System.out.println(year + " is not a leap year.");
        }
        else
            System.out.println(year + " is a leap year.");
    }
    else
        System.out.println(year + " is not a leap year.");
}
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);

        int year = sc.nextInt();
        leapYear(year);
    }
}
